### ISSUE TITLE
Too many iterations when fine-tuning a model a second time

### ISSUE THREAD

**Issue by woctezuma** [minimaxir/gpt-2-simple#13]
I have fine-tuned the GPT-2 model a first time with:

```python
sess = gpt2.start_tf_sess()

gpt2.finetune(sess,
              run_name=run_name,
              dataset=file_name,
              steps=1000,
              restore_from='fresh',   # change to 'latest' to resume training
              print_every=10,   # how many steps between printing progress
              sample_every=200,   # how many steps to print a demo sample
              save_every=500   # how many steps between saving checkpoint              
              )
```

I want to fine-tune it a second time, i.e. for 1000 additional iterations.
I have changed the cell to the following and executed it:

```python
sess = gpt2.start_tf_sess()

gpt2.finetune(sess,
              run_name=run_name,
              dataset=file_name,
              steps=1000,
              restore_from='latest',   # change to 'latest' to resume training
              print_every=10,   # how many steps between printing progress
              sample_every=200,   # how many steps to print a demo sample
              save_every=500   # how many steps between saving checkpoint              
              )
```

It seems to have started at 1001 iterations, which is as intended.
However, the process has not yet stopped. I thought it would stop at 2000 iterations.

```
Training...
[1010 | 32.03] loss=2.62 avg=2.62
[1020 | 56.97] loss=2.44 avg=2.53
[1030 | 81.37] loss=2.72 avg=2.59
[1040 | 106.37] loss=2.42 avg=2.55
[1050 | 131.60] loss=2.59 avg=2.56
[...]
[1970 | 2458.54] loss=2.03 avg=2.45
[1980 | 2483.27] loss=2.31 avg=2.45
[1990 | 2507.96] loss=2.27 avg=2.45
Saving checkpoint/descriptions/model-2000
[...]
[2000 | 2546.75] loss=2.47 avg=2.45
[2010 | 2571.65] loss=2.35 avg=2.45
[2020 | 2596.52] loss=2.37 avg=2.44
[2030 | 2621.22] loss=2.55 avg=2.45
[...]
[2960 | 4968.95] loss=2.31 avg=2.39
[2970 | 4993.75] loss=2.22 avg=2.39
[2980 | 5018.52] loss=2.48 avg=2.39
[2990 | 5043.32] loss=2.60 avg=2.39
Saving checkpoint/descriptions/model-3000
[...]
[3000 | 5081.95] loss=2.45 avg=2.39
[3010 | 5106.82] loss=2.48 avg=2.39
[3020 | 5131.65] loss=1.87 avg=2.39
[3030 | 5156.36] loss=2.23 avg=2.38
[3040 | 5181.04] loss=2.30 avg=2.38
```

**Comment by woctezuma** [comment]
Alright, the trick is that `steps` is the total number of steps. When fine-tuning a model, the parameter should be strictly higher than the number of steps previously done. Otherwise, the algorithm never terminates.

I should have written:

```python
sess = gpt2.start_tf_sess()

gpt2.finetune(sess,
              run_name=run_name,
              dataset=file_name,
              steps=2000,
              restore_from='latest',   # change to 'latest' to resume training
              print_every=10,   # how many steps between printing progress
              sample_every=200,   # how many steps to print a demo sample
              save_every=500   # how many steps between saving checkpoint              
              )
```

**Comment by minimaxir** [comment]
Fixed in 0.3.1

**Comment by yissachar** [comment]
@minimaxir Can you clarify which part is fixed? On 0.5.3 I still see that steps is the total number of steps, not the number of new steps as I expected.
